src/diceBot/NightWizard.rb

```diff
@@ -22,159 +24,181 @@ class NightWizard < DiceBot
 　例）12NW-5@7#2　　1NW　　50nw+5@7,10#2,5　50nw-5+10@7,10#2,5+15+25
 INFO_MESSAGE_TEXT
 
-  setPrefixes(['\d+NW'])
+  setPrefixes(['([-+]?\d+)?NW.*', '2R6.*'])
 
   def initialize
     super
     @sendMode = 2
+    @nw_command = "NW"
   end
 
-  def changeText(string)
-    return string unless string =~ /NW/i
-
-    string = string.gsub(/([\-\d]+)NW([\+\-\d]*)@([,\d]+)#([,\d]+)([\+\-\d]*)/i) do
-      modify = Regexp.last_match(5).empty? ? "" : ",#{Regexp.last_match(5)}"
-      "2R6m[#{Regexp.last_match(1)}#{Regexp.last_match(2)}#{modify}]c[#{Regexp.last_match(3)}]f[#{Regexp.last_match(4)}]"
+  def rollDiceCommand(string)
+    cmd = parse_nw(string) || parse_2r6(string)
+    unless cmd
+      return nil
     end
 
-    string = string.gsub(/([\-\d]+)NW([\+\-\d]*)/i) { "2R6m[#{Regexp.last_match(1)}#{Regexp.last_match(2)}]" }
-    string = string.gsub(/NW([\+\-\d]*)/i) { "2R6m[0#{Regexp.last_match(1)}]" }
+    total, interim_expr, status = roll_nw(cmd)
+    result =
+      if cmd.cmp_op
+        total.send(cmd.cmp_op, cmd.target_number) ? "成功" : "失敗"
+      end
+
+    sequence = [
+      "(#{cmd})",
+      interim_expr,
+      status,
+      total.to_s,
+      result,
+    ].compact
+    return sequence.join(" ＞ ")
   end
 
-  def dice_command_xRn(string, nick_e)
-    return checkRoll(string, nick_e)
+  private
+
+  class Parsed
+    # @!attribute critical_numbers
+    #   @return [Array<Integer>] クリティカルになる出目の一覧
+    # @!attribute fumble_numbers
+    #   @return [Array<Integer>] ファンブルになる出目の一覧
+    # @!attribute active_modify_number
+    #   @return [Integer] ファンブルでない時に適用される修正値
+    # @!attribute cmp_op
+    #   @return [Symbol, nil] 比較演算子
+    # @!attribute target_number
+    #   @return [Integer, nil] 目標値
+    attr_accessor :critical_numbers, :fumble_numbers, :active_modify_number, :cmp_op, :target_number
   end
 
-  def checkRoll(string, nick_e)
-    debug('checkRoll string', string)
-
-    output = '1'
-
-    num = '[,\d\+\-]+'
-    return output unless /(^|\s)S?(2R6m\[(#{num})\](c\[(#{num})\])?(f\[(#{num})\])?(([>=]+)(\d+))?)(\s|$)/i =~ string
-
-    debug('is valid string')
-
-    string = Regexp.last_match(2)
-    base_and_modify = Regexp.last_match(3)
-    criticalText = Regexp.last_match(4)
-    criticalValue = Regexp.last_match(5)
-    fumbleText = Regexp.last_match(6)
-    fumbleValue = Regexp.last_match(7)
-    judgeText = Regexp.last_match(8)
-    judgeOperator = Regexp.last_match(9)
-    judgeValue = Regexp.last_match(10).to_i
-
-    crit = "0"
-    fumble = "0"
-    cmp_op = nil
-    diff = 0
+  class ParsedNW < Parsed
+    # @!attribute base
+    #   @return [Integer] 判定の基礎値
+    # @!attribute modify_number
+    #   @return [Integer] 修正値
+    attr_accessor :base, :modify_number
```

Comment:
上と同様に、`@!attribute` を使わずに単純化できます。

```suggestion
    # @return [Integer] 判定の基礎値
    attr_accessor :base

    # @return [Integer] 修正値
    attr_accessor :modify_number
```